### PR TITLE
Remove warnings at the root

### PR DESCRIPTION
**Pull request by SamFerracin**
**Description of the Change:**
Removes ``ComplexWarning`` at the root (inside the backend) rather than in the functions that require casting

### CODE REVIEW

**Review by sduquemesa — APPROVED** [review]
Thanks @SamFerracin! 

I wonder if this PR is also a good opportunity to disable tensorflow's CUDA warning messages — those that appear when importing TF and no GPU is available. What do you think? One option is to just disable them and the other is to install `tensorflow-cpu` instead of the full `tensorflow` package that includes GPU support.

**Comment by SamFerracin** [comment]
@sduquemesa That's a great idea. i hate those warnings. Let me look into it :)

File: mrmustard/math/backend_tensorflow.py [hunk 1]
```diff
@@ -87,6 +87,9 @@ def boolean_mask(self, tensor: tf.Tensor, mask: tf.Tensor) -> Tensor:
     def cast(self, array: tf.Tensor, dtype=None) -> tf.Tensor:
         if dtype is None:
             return array
+
+        if dtype not in [self.complex64, self.complex128, "complex64", "complex128"]:
+            array = self.real(array)
```
Comment (ziofil) [inline mrmustard/math/backend_tensorflow.py:92]: what if we cast to int32?

**Review by ziofil — APPROVED** [review]
approved, up to you to cover casting to int32 correctly

File: mrmustard/math/backend_tensorflow.py (hunk 1 above)
Comment (SamFerracin) [reply mrmustard/math/backend_tensorflow.py:92]: I only see warnings when I convert from complex to something else, and this case is covered.
Do you have a specific example in mind that is not covered?

**Pull request merged**